### PR TITLE
Refresh the two key vault templates

### PR DESCRIPTION
**Pull request by mumian**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by mumian** [comment]
@bmoore-msft - Hi Brian, not sure what is missing with the PR.  The error says "template validation failed".  However, I was able to deploy the template from my computer.  Thanks.

File: 101-key-vault-create/azuredeploy.json [hunk 1]
```diff
@@ -5,118 +5,135 @@
...
-    "enableVaultForDiskEncryption": {
-      "type": "bool",
-      "defaultValue": false,
-      "allowedValues": [
-        true,
-        false
-      ],
+    "secretName": {
+      "type": "string",
+      "defaultValue": "vmAdminPassword",
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: remove the defaultValue

File: 101-key-vault-create/azuredeploy.json [hunk 2]
```diff
@@ -5,118 +5,135 @@
...
-      "type": "string",
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
-        "description": "Location for all resources."
+        "description": "Value of the secret that you want to create."
       }
     }
   },
+  "variables": {
+    "enabledForDeployment": false,
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: revert this change, these should be parameters

File: 101-key-vault-create/azuredeploy.json [hunk 3]
```diff
@@ -5,118 +5,135 @@
...
         "accessPolicies": [
           {
-            "tenantId": "[parameters('tenantId')]",
             "objectId": "[parameters('objectId')]",
+            "tenantId": "[parameters('tenantId')]",
             "permissions": {
-              "keys": "[parameters('keysPermissions')]",
-              "secrets": "[parameters('secretsPermissions')]"
+              "keys": [
+                "Get",
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: revert this change, these should be params

File: 101-key-vault-create/azuredeploy.json [hunk 4]
```diff
@@ -5,118 +5,135 @@
...
+    "enabledForDeployment": false,
+    "enabledForDiskEncryption": false,
+    "enabledForTemplateDeployment": false,
+    "sku": "Standard"
+  },
   "resources": [
     {
       "type": "Microsoft.KeyVault/vaults",
       "name": "[parameters('keyVaultName')]",
       "apiVersion": "2016-10-01",
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: update apiVersion

File: 101-key-vault-create/azuredeploy.json [hunk 5]
```diff
@@ -5,118 +5,135 @@
...
+            "virtualNetworkRules": [],
+            "ipRules": []
+          }
+        }
+      }
+    },
+    {
+      "type": "Microsoft.KeyVault/vaults/secrets",
+      "name": "[concat(parameters('keyVaultName'), '/', parameters('secretName'))]",
+      "apiVersion": "2016-10-01",
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: update apiVersion

File: 101-key-vault-create/azuredeploy.json [hunk 6]
```diff
@@ -5,118 +5,135 @@
...
+      "location": "[parameters('location')]",
+      "scale": null,
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: remove this

File: 101-key-vault-create/azuredeploy.json [hunk 7]
```diff
@@ -5,118 +5,135 @@
...
+      "dependsOn": [
+        "[resourceId('Microsoft.KeyVault/vaults', parameters('keyVaultName'))]"
+      ],
+      "properties": {
+        "contentType": "securestring",
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: remove this

File: 101-key-vault-create/azuredeploy.json [hunk 8]
```diff
@@ -5,118 +5,135 @@
...
+        "value": "[parameters('secretValue')]",
+        "attributes": {
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: remove this

File: 101-key-vault-create/azuredeploy.json [hunk 9]
```diff
@@ -5,118 +5,135 @@
...
         "sku": {
-          "name": "[parameters('skuName')]",
+          "name": "[variables('sku')]",
           "family": "A"
+        },
+        "networkAcls": {
+          "value": {
+            "defaultAction": "Allow",
+            "bypass": "AzureServices",
+            "virtualNetworkRules": [],
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.json]: remove the empty arrays

File: 101-key-vault-create/azuredeploy.parameters.json [hunk 10]
```diff
@@ -3,10 +3,13 @@
   "contentVersion": "1.0.0.0",
   "parameters": {
     "keyVaultName": {
-      "value": "GEN-UNIQUE"
+      "value": "GEN-KEYVAULT-NAME"
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.parameters.json]: use GEN-UNIQUE, what you have is used for referencing the existing vault

File: 101-key-vault-create/azuredeploy.parameters.json [hunk 11]
```diff
@@ -3,10 +3,13 @@
...
+    },
+    "tenantId": {
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.parameters.json]: remove this since you have a defaultValue

File: 101-key-vault-create/azuredeploy.parameters.json [hunk 12]
```diff
@@ -3,10 +3,13 @@
...
+      "value": "GEN-GUID"
     },
     "objectId": {
-      "value": "GEN-AAD-OBJECTID"
+      "value": "GEN-GUID"
```
Comment (bmoore-msft) [inline 101-key-vault-create/azuredeploy.parameters.json]: use GEN-AZURE_OBJECT_ID

File: 201-key-vault-secret-create/azuredeploy.json [hunk 13]
```diff
@@ -2,107 +2,133 @@
   "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
   "contentVersion": "1.0.0.0",
   "parameters": {
-    "keyVaultName": {
+    "name": {
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: revert this change

File: 201-key-vault-secret-create/azuredeploy.json [hunk 14]
```diff
@@ -2,107 +2,133 @@
...
-      "type": "string",
-      "defaultValue": "[resourceGroup().location]",
-      "metadata": {
-        "description": "Location for all resources."
+        "description": "all secrets {\"secretName\":\"\",\"secretValue\":\"\"} wrapped in a secure object."
       }
     }
   },
+  "variables": {
+    "enabledForDeployment": false,
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: revert this change, these should be parameters

File: 201-key-vault-secret-create/azuredeploy.json [hunk 15]
```diff
@@ -2,107 +2,133 @@
...
+        "enabledForTemplateDeployment": "[variables('enabledForTemplateDeployment')]",
+        "enabledForDiskEncryption": "[variables('enabledForDiskEncryption')]",
         "tenantId": "[parameters('tenantId')]",
-        "accessPolicies": "[parameters('accessPolicies')]",
+        "accessPolicies": [
+          {
+            "objectId": "[parameters('objectId')]",
+            "tenantId": "[parameters('tenantId')]",
+            "permissions": {
+              "keys": [
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: revert this change these should be parameters

File: 201-key-vault-secret-create/azuredeploy.json [hunk 16]
```diff
@@ -2,107 +2,133 @@
...
         "sku": {
-          "name": "[parameters('vaultSku')]",
+          "name": "[variables('sku')]",
           "family": "A"
+        },
+        "networkAcls": {
+          "value": {
+            "defaultAction": "Allow",
+            "bypass": "AzureServices",
+            "virtualNetworkRules": [],
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: remove empty arrays

File: 201-key-vault-secret-create/azuredeploy.json [hunk 17]
```diff
@@ -2,107 +2,133 @@
...
+      "dependsOn": [
+        "[concat('Microsoft.KeyVault/vaults/', parameters('name'))]"
+      ],
       "properties": {
         "value": "[parameters('secretsObject').secrets[copyIndex()].secretValue]"
       },
-      "dependsOn": [
-        "[concat('Microsoft.KeyVault/vaults/', parameters('keyVaultName'))]"
-      ],
       "copy": {
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.json]: this loop needs an input property

File: 201-key-vault-secret-create/azuredeploy.parameters.json [hunk 18]
```diff
@@ -1,46 +1,29 @@
...
-            }
-        }
-    }
\ No newline at end of file
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "name": {
+      "value": "GEN-KEYVAULT-NAME"
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.parameters.json]: GEN-UNIQUE

File: 201-key-vault-secret-create/azuredeploy.parameters.json [hunk 19]
```diff
@@ -1,46 +1,29 @@
...
+    },
+    "tenantId": {
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.parameters.json]: remove this

File: 201-key-vault-secret-create/azuredeploy.parameters.json [hunk 20]
```diff
@@ -1,46 +1,29 @@
...
+      "value": "GEN-GUID"
+    },
+    "objectId": {
+      "value": "GEN-GUID"
```
Comment (bmoore-msft) [inline 201-key-vault-secret-create/azuredeploy.parameters.json]: GEN-AZURE_OBJECT_ID

File: 101-key-vault-create/azuredeploy.json (hunk 2 above)
Comment (mumian) [reply 101-key-vault-create/azuredeploy.json]: Done.  I also move the other two enabled values to the parameter section.

File: 101-key-vault-create/azuredeploy.json (hunk 4 above)
Comment (mumian) [reply 101-key-vault-create/azuredeploy.json]: Done

File: 201-key-vault-secret-create/azuredeploy.json (hunk 17 above)
Comment (mumian) [reply 201-key-vault-secret-create/azuredeploy.json]: Not sure I understand.  The input property is an object of a list secrets.

Comment (bmoore-msft) [reply 201-key-vault-secret-create/azuredeploy.json]: nm, I see it... move "copy" above "properties"...
